go-controller/cmd/ovn-kube-gateway-helper/ovn-kube-gateway-helper.go

```diff
@@ -0,0 +1,328 @@
+	stdout, _, err = util.RunOVSOfctl(
+		"add-flow", physicalBridge,
+		fmt.Sprintf("priority=100, table=1, ct_state=+trk+est, actions=output:%d", brIntOfport))
+	if err != nil {
+		return fmt.Errorf("table 1, add flow for established and related connections go to pod failed: %q", stdout)
+	}
+
+	// All other connections get the 'NORMAL' treatment
+	stdout, _, err = util.RunOVSOfctl(
+		"add-flow", physicalBridge,
+		"priority=0, table=1, actions=NORMAL")
+	if err != nil {
+		return fmt.Errorf("table 1, add flow for other connections failed: %q", stdout)
+	}
+
+	return nil
+}
+
+func syncServices(kubecli *kube.Kube, physicalBridge string, physicalInterfaceOfport int) error {
+	existingNamespaces, err := kubecli.GetNamespaces()
+	if err != nil {
+		return fmt.Errorf("get namespaces failed: %v", err)
+	}
+
+	nodePorts := make(map[string]bool)
+	for _, namespace := range existingNamespaces.Items {
+		var existingServices *kapi.ServiceList
+
+		existingServices, err = kubecli.GetServices(namespace.ObjectMeta.Name)
+		if err != nil {
+			return fmt.Errorf("get services failed: %v", err)
+		}
+
+		for _, service := range existingServices.Items {
+			if service.Spec.Type != kapi.ServiceTypeNodePort || len(service.Spec.Ports) == 0 {
+				continue
+			}
+
+			for _, svcPort := range service.Spec.Ports {
+				port := svcPort.NodePort
+				if port == 0 {
+					continue
+				}
+
+				prot := svcPort.Protocol
+				if prot != TCP && prot != UDP {
+					continue
+				}
+				protocol := strings.ToLower(string(prot))
+				nodePortKey := fmt.Sprintf("%s_%d", protocol, port)
+				nodePorts[nodePortKey] = true
+			}
+		}
+	}
+
+	// Possibility that the daemon was restarted. We should find out any additional
+	// node_ports which are stale and delete them
+	stdout, _, err := util.RunOVSOfctl(
+		"dump-flows", physicalBridge,
+		fmt.Sprintf("in_port=%d", physicalInterfaceOfport))
+	if err != nil {
+		return fmt.Errorf("dump flows failed: %q", stdout)
+	}
+	flows := strings.Split(stdout, "\n")
+
+	re, err := regexp.Compile(`tp_dst=(.*?)[, ]`)
+	if err != nil {
+		return fmt.Errorf("regexp compile failed: %v", err)
+	}
+
+	for _, flow := range flows {
+		group := re.FindStringSubmatch(flow)
+		if group == nil {
+			continue
+		}
+
+		var key string
+		if strings.Contains(flow, "tcp") {
+			key = fmt.Sprintf("tcp_%s", group[1])
+		} else if strings.Contains(flow, "udp") {
+			key = fmt.Sprintf("udp_%s", group[1])
+		} else {
+			continue
+		}
+
+		if _, ok := nodePorts[key]; !ok {
+			pair := strings.Split(key, "_")
+			protocol, port := pair[0], pair[1]
+			protocolDst := fmt.Sprintf("%s_dst", protocol)
+
+			stdout, _, err := util.RunOVSOfctl(
+				"del-flows", physicalBridge,
+				fmt.Sprintf("in_port=%d, %s, %s=%s",
+					physicalInterfaceOfport, protocol, protocolDst, port))
+			if err != nil {
+				logrus.Errorf("del flow of %s failed: %q", physicalBridge, stdout)
+			}
+		}
+	}
+
+	return nil
+}
+
+func main() {
```

Comment:
Please break this function up. The main function is too long. Handle the parse args part separately from the logic maybe.